tests/test_dht_crypto.py

```diff
@@ -41,3 +45,88 @@ def test_rsa_signature_validator():
                        for record in protected_records]  # With someone else's signature
     for record in signed_records:
         assert not receiver_validator.validate(record)
+
+
+def test_cached_key():
+    first_validator = RSASignatureValidator()
+    second_validator = RSASignatureValidator()
+    assert first_validator.local_public_key == second_validator.local_public_key
+
+    third_validator = RSASignatureValidator(ignore_cached_key=True)
+    assert first_validator.local_public_key != third_validator.local_public_key
+
+
+def test_validator_instance_is_pickable():
+    # Needs to be pickable because the validator instance may be sent between processes
```

Comment:
```suggestion
def test_validator_instance_is_picklable():
    # Needs to be picklable because the validator instance may be sent between processes
```

Reply:
Also check all other occurences